### ISSUE TITLE
[MASSEMBLY-494] No way to set the directory mode for the base directory nor any implicitly created directory for zip assemblies

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-assembly-plugin#663]
**[zobule](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=zobule)** opened **[MASSEMBLY-494](https://issues.apache.org/jira/browse/MASSEMBLY-494?redirect=false)** and commented

A sample maven project is attached to reproduce everything. I am trying to generate a zip file with correct permissions (i.e. directories not writable by everyone which I would say should not be too difficult).

If you run `mvn assembly:assembly` as is, you will have:

drwxrwxrwx ... assembly-test-1.0.0/
drwxrwxrwx ... assembly-test-1.0.0/bin/
-rw-r--r-- ... assembly-test-1.0.0/bin/assembly-test-1.0.0.jar
-rw-r--r-- ... assembly-test-1.0.0/bin/commons-lang-2.4.jar
drwxr-xr-x ... assembly-test-1.0.0/copyofsubdir/
-rw-r--r-- ... assembly-test-1.0.0/copyofsubdir/file

I found no way to set the first two directories permissions. If you add the two directoryMode tags, you will have:

drwxrwxrwx ... assembly-test-1.0.0/
drwxrwxrwx ... assembly-test-1.0.0/bin/
-rw-r--r-- ... assembly-test-1.0.0/bin/assembly-test-1.0.0.jar
-rw-r--r-- ... assembly-test-1.0.0/bin/commons-lang-2.4.jar
drwxr-xr-x ... assembly-test-1.0.0/copyofsubdir/
-rwxrwxrwx ... assembly-test-1.0.0/copyofsubdir/file*

... which is pretty much what I did NOT expect (a cut'n paste issue in the code?). If you add both directoryMode and fileMode, you will have the same results as the first iteration (see above).

Unfortunately I still cannot change the permissions of:

    - the base directory (assembly-test-1.0.0)
    - any directory that is not present on the file system (bin)

I looked a bit at the code and, as far as I can see, it may be linked to the plexus zip archiver (although it may not be responsible for the bug) which automatically adds parent directories but most probably with incorrect permissions due to bad configuration.

I did not test with other assembly formats.

PS: yes I can do a simple tarball, and I will probably do that while waiting for a patch. But I think it is important that you are aware of something which may annoy more than one single user in the world.

Regards.

---

**Affects:** 2.2-beta-5

**Attachments:**
- [assembly-test.zip](https://issues.apache.org/jira/secure/attachment/12714187/assembly-test.zip) (_1.75 kB_)

**Comment by jira-importer** [comment]
**[John Dennis Casey](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=jdcasey)** commented

Try the following assembly plugin configuration, in your pom.xml:

```xml
<archiverConfig>
  <defaultDirectoryMode>0755</defaultDirectoryMode>
</archiverConfig>
```

I've verified that this provides the ability to manage the default directory mode for automatically created / implied directories.

**Comment by jira-importer** [comment]
**[zobule](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=zobule)** commented

It works, but I have the 493 issue which I suspect to be a bug (affects 2.3 too).

And you did not mention the fact that in 2.2-beta-5, the directoryMode tag changed the permissions of files instead of directories. Was not a bug? Fortunately it seems to be fixed in 2.3.
